src/datagrid/DataGrid.tsx

```diff
@@ -0,0 +1,396 @@
+                ])}
+            >
+                <span className={ClassNames.DATAGRID_COLUMN_TITLE}>
+                    {selectionType === GridSelectionType.MULTI && (
+                        <CheckBox
+                            onChange={evt => this.handleSelectAll(evt)}
+                            ariaLabel="Select All"
+                            checked={selectAll !== undefined ? selectAll : undefined}
+                        />
+                    )}
+                </span>
+                <div className={ClassNames.DATAGRID_COLUMN_SEPARATOR} />
+            </div>
+        );
+    }
+
+    // function to render select cell
+    private buildSelectCell(rowID: string, isSelected: any): React.ReactElement {
+        const {selectionType} = this.props;
+        if (selectionType === GridSelectionType.MULTI) {
+            return (
+                <CheckBox
+                    id={rowID}
+                    ariaLabel="Select"
+                    onChange={evt => this.handleSelectSingle(evt, rowID)}
+                    checked={isSelected !== undefined ? isSelected : undefined}
+                />
+            );
+        }
+        return (
+            <RadioButton
+                value={rowID}
+                id={rowID}
+                onChange={evt => this.handleSelectSingle(evt, rowID)}
+                checked={isSelected !== undefined ? isSelected : undefined}
+            />
+        );
+    }
+
+    // function to build datagrid body
+    private buildDataGridBody(): React.ReactElement {
+        const {data} = this.props;
+        return (
+            <div className={ClassNames.DATAGRID}>
+                <div className={ClassNames.DATAGRID_TABLE_WRAPPER}>
+                    <div className={ClassNames.DATAGRID_TABLE} role="grid">
+                        {this.buildDataGridHeader()}
+                        {data ? (
+                            data.map((row: any, index: number) => {
+                                return this.buildDataGridRow(
+                                    row.rowID,
+                                    row.content,
+                                    index,
+                                    row.className,
+                                    row.style,
+                                    row.isSelected,
+                                );
+                            })
+                        ) : (
+                            <div className={ClassNames.DATAGRID_PLACEHOLDER_CONTAINER}>
+                                <div className={ClassNames.DATAGRID_PLACEHOLDER} />
+                            </div>
+                        )}
+                    </div>
+                </div>
+            </div>
+        );
+    }
+
+    // Function to build datagrid header
+    private buildDataGridHeader(): React.ReactElement {
+        const {columns, selectionType} = this.props;
+        return (
+            <div className={ClassNames.DATAGRID_HEADER} role="rowgroup">
```

Comment:
Checking VMWare Clarity website, this div should get an additional class `datagrid-selected` when the checkbox is selected.

Reply:
@ben-schumacher  Thanks for noticing this. the 'datagrid-selected' gets added to row when row is selected.